Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1321,83 +1321,246 @@ auto RenderTheme::colorCache(OptionSet<StyleColorOptions> options) const -> Colo
 Color RenderTheme::systemColor(CSSValueID cssValueId, OptionSet<StyleColorOptions> options) const
 {
     switch (cssValueId) {
-    case CSSValueWebkitLink:
-        return options.contains(StyleColorOptions::ForVisitedLink) ? SRGBA<uint8_t> { 85, 26, 139 } : SRGBA<uint8_t> { 0, 0, 238 };
-    case CSSValueWebkitActivelink:
-    case CSSValueActivetext:
-        return Color::red;
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-canvas
+    // Background of application content or documents.
+    case CSSValueCanvas:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-canvastext
+    // Text in application content or documents.
+    case CSSValueCanvastext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-linktext
+    // Text in non-active, non-visited links. For light backgrounds, traditionally blue.
     case CSSValueLinktext:
         return SRGBA<uint8_t> { 0, 0, 238 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-visitedtext
+    // Text in visited links. For light backgrounds, traditionally purple.
     case CSSValueVisitedtext:
         return SRGBA<uint8_t> { 85, 26, 139 };
-    case CSSValueActiveborder:
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-activetext
+    // Text in active links. For light backgrounds, traditionally red.
+    case CSSValueActivetext:
+    case CSSValueWebkitActivelink: // Non-standard addition.
+        return Color::red;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonface
+    // The face background color for push buttons.
+    case CSSValueButtonface:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttontext
+    // Text on push buttons.
+    case CSSValueButtontext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonborder
+    // The base border color for push buttons.
+    case CSSValueButtonborder:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-field
+    // Background of input fields.
+    case CSSValueField:
         return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-fieldtext
+    // Text in input fields.
+    case CSSValueFieldtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlight
+    // Background of selected text, for example from ::selection.
+    case CSSValueHighlight:
+        return SRGBA<uint8_t> { 181, 213, 255 };
```

Comment:
I am a little bit confused; the old code does this same thing, so this is not a change. But macOS has a highlight color that is chosen in the System Settings app. Should this be exposed here? Seems like a fingerprinting risk, but not exposing it seems like a correctness risk.

Reply:
Indeed it would be a fingerprinting risk to expose the system color. To mitigate that and still allow native UI to use the accent derived colors we tied that to the WKWebView `_useSystemAppearance` SPI. When that is `NO` we return fixed blue derived colors only. When it is `YES` we use the `NSColor` / `UIColor` APIs to return real accent derived colors.

Reply:
> I am a little bit confused; the old code does this same thing, so this is not a change. But macOS has a highlight color that is chosen in the System Settings app. Should this be exposed here? Seems like a fingerprinting risk, but not exposing it seems like a correctness risk.

This is existing code, I just reformatted and re-ordered it.

As Tim notes, we actually do have an override of this in RenderThemeMac/RenderThemeIOS that does a more fingerprinty thing when a native might want it.